### PR TITLE
Create Daily Report with validate.

### PR DESCRIPTION
**Pull request by phuongthuan**

### CODE REVIEW

File: src/components/IssueSelect/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    {({ loading, error, data } ) => {
+      if (loading) return <div>Loading...</div>;
+      if (error) return <div>Error</div>;
+      const { issues } = data;
+
+      return (
+        <FieldArray
+          name="issues"
+          render={arrayHelpers => (
+            <div className="row mb-3">
```
Comment (queq1890) [inline src/components/IssueSelect/index.js]: Do you use these className selectors, `row`, `mb-3` `col-md-4` etc?

File: src/components/DailyReportForm/index.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+`;
+
+const DailyReportSchema = Yup.object().shape({
+  title: Yup.string()
+    .required('Title is required.'),
+  achievement: Yup.string()
+    .required('Achievement is required.'),
+  plan: Yup.string()
+    .required('Plan is required.'),
+  // issues: Yup.array()
```
Comment (queq1890) [inline src/components/DailyReportForm/index.js:42]: Any problem for issues validation?

File: src/components/TextArea/index.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          rows="5"
+          cols="30"
+          autoComplete="off"
+          value={value}
+          onChange={onChange}
+          {...props}
+        />
+      </Fragment>
+    )}
+    {error ? (
```
Comment (queq1890) [inline src/components/TextArea/index.js]: IMO: 
I think here can be `{ error && TextError>{error}</TextError> }`.

File: src/components/TextInput/index.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+          error={error}
+          type={type}
+          autoComplete="off"
+          value={value}
+          onChange={onChange}
+          {...props}
+        />
+      </Fragment>
+    )}
+    {error ? (
```
Comment (queq1890) [inline src/components/TextInput/index.js]: same above

**Review by queq1890 — APPROVED** [review]
LGTM with some small comments. I think you can merge it 🎉

File: src/components/DailyReportForm/index.js (hunk 2 above)
Comment (phuongthuan) [reply src/components/DailyReportForm/index.js:42]: Yeah, I'll add this later.

File: src/components/TextArea/index.js (hunk 3 above)
Comment (phuongthuan) [reply src/components/TextArea/index.js]: You're right. Fixed! Thank for mentioned.

File: src/components/AsyncButton/index.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { AsyncButtonStyles } from '../../styles/AsyncButton';
+
+const AsyncButton = ({ type, onClick, loading }) => {
```
Comment (ggianglv) [inline src/components/AsyncButton/index.js]: my be this better
```
const AsyncButton = ({ type, onClick, loading }) => (
  <AsyncButtonStyles
        disabled={loading}
        type={type}
        onClick={onClick}
      >
        {loading ? 'Submitting...' : 'Submit'}
      </AsyncButtonStyles>
)
```

Comment (phuongthuan) [reply src/components/AsyncButton/index.js]: @gianglevan94 Yeah you're right. Thanks 💯

File: src/components/DailyReportForm/index.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+            description: '',
+            comment: ''
+          }}
+          enableReinitialize
+          validationSchema={DailyReportSchema}
+          onSubmit={async (values, { setSubmitting, setStatus, setErrors, resetForm }) => {
+            const { title, achievement, plan, description, comment } = values;
+
+            let issues = [];
+            if (isEmpty(values.issues)) {
```
Comment (ggianglv) [inline src/components/DailyReportForm/index.js]: better 
```
if(!isEmpty(values.issues)) {
  issues = values.issues.map(issue => ({ id: issue }));
}
```

or `const issues = isEmpty(values.issues) ? [] : values.issues.map(issue => ({ id: issue }));`

Comment (phuongthuan) [reply src/components/DailyReportForm/index.js]: @gianglevan94 I like the second way, thanks

**Review by ggianglv — APPROVED** [review]

**Review by queq1890 — APPROVED** [review]

**Pull request merged**